adding 3 variables to fix Pre shared infrastructure

SalimKainos opened the pull request:
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:


### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSPO-14094

### Change description ###



**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

SalimKainos committed:
adding 3 variables to fix Pre shared infrastructure

github-actions[bot] committed:
terraform-docs: automated action

thomast1906 commented on inputs-optional.tf:
```diff
@@ -36,7 +36,7 @@ variable "ipconfig_name" {
   type        = string
   description = "The type of private IP allocation, either Static or Dynamic."
-  default     = "Static"
+  default     = "Dynamic"
```
why setting to dynamic ?

TomKimber replied:
This was discussed with @timja. We both agree in that in most use cases the consumer of the module wouldn't be concerned with what IP was assigned so this should be the default but that it should be overridable if required.

SalimKainos committed:
Back to static

SalimKainos committed:
Merge branch 'fix-pre-shared' of https://github.com/hmcts/terraform-module-virtual-machine into fix-pre-shared

github-actions[bot] committed:
terraform-docs: automated action

timja reviewed (commented):
why is this PR needed?

SalimKainos commented:
> why is this PR needed?


<img width="1236" alt="Screenshot 2023-06-09 at 09 08 54" src="https://github.com/hmcts/terraform-module-virtual-machine/assets/102227453/db36651c-334f-41ff-bf65-afa6f1328b58">

> why is this PR needed?

Zoe asked me to revert the changes back. Pipeline is failing.

timja commented:
when was this changed though? the PR you filed didn't touch variables to do with that: https://github.com/hmcts/terraform-module-virtual-machine/pull/27

SalimKainos commented:
> when was this changed though? the PR you filed didn't touch variables to do with that: #27

The pipeline was failing because someone before me made some changes to the code. I tried to fix that.

timja commented:
> > when was this changed though? the PR you filed didn't touch variables to do with that: #27
> 
> The pipeline was failing because someone before me made some changes to the code. I tried to fix that.

this just adds dead code which doesn't do anything. Why don't you remove it from the pre repository?

SalimKainos commented:
> > > when was this changed though? the PR you filed didn't touch variables to do with that: #27
> > 
> > 
> > The pipeline was failing because someone before me made some changes to the code. I tried to fix that.
> 
> this just adds dead code which doesn't do anything. Why don't you remove it from the pre repository?

I removed from Pre shared infrastructure not from here. To fix the pipeline I removed that.

nchang90 commented:
Is this still required as it is 2 years ago @SalimKainos
